lib/SPIRV/SPIRVReader.cpp

```diff
@@ -4825,6 +4825,137 @@ Instruction *SPIRVToLLVM::transRelational(SPIRVInstruction *I, BasicBlock *BB) {
   return cast<Instruction>(Mutator.getMutated());
 }
 
+std::optional<SPIRVModuleReport> getSpirvReport(std::istream &IS) {
+  int IgnoreErrCode;
+  return getSpirvReport(IS, IgnoreErrCode);
+}
+
+std::optional<SPIRVModuleReport> getSpirvReport(std::istream &IS,
+                                                int &ErrCode) {
+  SPIRVWord Word;
+  std::string Name;
+  std::unique_ptr<SPIRVModule> BM(SPIRVModule::createSPIRVModule());
+  SPIRVDecoder D(IS, *BM);
+  D >> Word;
+  if (Word != MagicNumber) {
+    ErrCode = SPIRVEC_InvalidMagicNumber;
```

Comment:
Should we try to return the invalid data (in all possible cases) as well? Or will that be redundant?

Thanks

Reply:
A general scheme is to validate the SPIRV header and either return valid data & success code, or return empty std::optional with a properly set error code.

Reply:
To clarify, I was thinking if we need to report the error as follows:
'5' is not a valid magic number.
But, reading your earlier comment about the error reporting not being the primary feature, I suppose it's not that important. Thanks